cranelift/codegen/meta/src/gen_asm.rs

```diff
@@ -117,6 +126,17 @@ pub fn generate_macro_inst_fn(f: &mut Formatter, inst: &Inst) {
                 [] => fmtln!(f, "SideEffectNoResult::Inst(inst)"),
                 [one] => match one.mutability {
                     Read => unreachable!(),
+                    Write => match one.location.kind() {
+                        // One write-only register output? Output the
+                        // instruction and that register.
+                        OperandKind::Reg(r) => {
+                            let ty = r.reg_class().unwrap().to_string();
+                            let var = ty.to_lowercase();
+                            fmtln!(f, "let {var} = {r}.as_ref().clone();");
+                            fmtln!(f, "AssemblerOutputs::Ret{ty} {{ inst, {var} }}");
+                        }
+                        _ => unimplemented!(),
+                    },
```

Comment:
Should this get collapsed into `ReadWrite` as well below? This has the same handling for `Reg` and all the other cases look applicable for one-operand-written instructions too (e.g. if something stores in memory or outputs to a fixed register)

Reply:
Unfortunately, there's slightly different behavior for retrieving the underlying Cranelift type. The `ReadWrite` variant needs to access the `write` field; this one should actually be doing a `as_ref().to_reg()` (let me fix that). I think in the fullness of time we will want to merge some things like I did over in the assembler generator but let's leave this as-is until we have a few more write-only instructions implemented.